src/filelock/_api.py

```diff
@@ -143,6 +136,19 @@ def _initialize(  # noqa: PLR0913
             to pass the same object around.
 
         """
+        if is_singleton and hasattr(self, "_context"):
+            # test whether other parameters match existing instance.
+            if (
+                self.is_thread_local() != thread_local
+                or not self.is_singleton
+                or self._context.timeout != timeout
+                or self._context.mode != mode
+                or self._context.blocking != blocking
+            ):
+                msg = "Singleton lock instances cannot be initialized with differing arguments"
```

Comment:
this message is unfortunate - it doesn't tell the reader anything about the differences.

Reply:
Yeah this can be expanded to give more details. Will update.

Reply:
Complex enough to need a test now.

Reply:
Actually there is a test for it, which I will expand to include the other parameters, but it doesn't seem to mind that the error message has changed - guess that's not considered enough to fail a test? 

```python
with pytest.raises(ValueError, match="Singleton lock instances cannot be initialized with differing arguments"):
        lock_type(str(lock_path), timeout=10, is_singleton=True)
    with pytest.raises(ValueError, match="Singleton lock instances cannot be initialized with differing arguments"):
        lock_type(str(lock_path), mode=0, is_singleton=True)
```

Reply:
Oh never mind, I see that it doesn't have to be a full string match.